packages/pricing/src/services/pricing-module.ts

```diff
@@ -1750,11 +1775,49 @@ export default class PricingModuleService<
     data: PricingTypes.AddPriceListPricesDTO[],
     sharedContext: Context = {}
   ): Promise<PricingTypes.PriceListDTO[]> {
+    const ruleTypeAttributes: string[] = []
+    const priceListIds: string[] = []
+
+    for (const priceListData of data) {
+      priceListIds.push(priceListData.priceListId)
+
+      for (const price of priceListData.prices) {
+        ruleTypeAttributes.push(...Object.keys(price.rules || {}))
+      }
+    }
+
+    const ruleTypes = await this.listRuleTypes(
+      { rule_attribute: ruleTypeAttributes },
+      { take: null }
+    )
+
+    const ruleTypeMap: Map<string, RuleTypeDTO> = new Map(
+      ruleTypes.map((rt) => [rt.rule_attribute, rt])
+    )
+
+    for (const ruleAttribute of arrayUnique<string>(ruleTypeAttributes)) {
+      let ruleType = ruleTypeMap.get(ruleAttribute)
+
+      if (ruleType) {
+        continue
+      }
+
+      ;[ruleType] = await this.createRuleTypes(
+        [
+          {
+            name: ruleAttribute,
+            rule_attribute: ruleAttribute,
+          },
+        ],
+        sharedContext
+      )
+
+      ruleTypeMap.set(ruleAttribute, ruleType)
+    }
+
     const priceLists = await this.listPriceLists(
-      { id: data.map((d) => d.priceListId) },
-      {
-        take: null,
-      },
+      { id: priceListIds },
+      { take: null },
```

Comment:
We should validate this change first, but buildQuery should now no longer add `limit: 15` by default if you are filtering using `id` filters, so this shouldn't be necessary anymore, but let's double-triple check before removing this
```suggestion
```